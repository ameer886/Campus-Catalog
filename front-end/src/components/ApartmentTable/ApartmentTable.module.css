.ApartmentTable {
  align-items: center;
  background-color: rgb(184, 181, 181);
  margin-top: 100px;
  font-family: 'Times New Roman', Times, serif;
}
