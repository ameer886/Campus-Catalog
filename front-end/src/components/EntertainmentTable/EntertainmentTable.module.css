.EntertainmentTable {
  align-items: center;
  background-color: rgb(230, 228, 228);
  margin-top: 100px;
  font-family: 'Times New Roman', Times, serif;
}

.FilterButton {
  float: right;
  margin: 8px;
}
