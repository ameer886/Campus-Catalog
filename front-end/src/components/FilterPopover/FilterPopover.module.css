.PopoverButton {
  background-color: white;
}

.IconAdjust {
  margin-top: -4px;
  color: black;
  width: 20px;
  height: 20px;
}

.InputDiv {
  display: flex;
  align-items: center;
  margin-left: 8px;
}

.Input {
  margin: 0 4px;
}

.TitleSplitter {
  display: flex;
  justify-content: space-between;
}

.OptionHeader {
  color: #555;
}
