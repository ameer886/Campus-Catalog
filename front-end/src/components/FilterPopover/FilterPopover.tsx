import React from 'react';

import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import Popover from 'react-bootstrap/Popover';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import { BsFilterCircle } from 'react-icons/bs';

import styles from './FilterPopover.module.css';

type FilterPopoverCheckboxOption = {
  header: string;
  key: string;
  variant: 'checkbox' | 'radio';
  boxValues: Array<{
    value: string;
    displayStr: string;
    // Generally __checked is for internal use only.
    // However, if you want a component to default to being checked,
    // you can set this to true.
    __checked?: boolean;
  }>;
};

type FilterPopoverInputOption = {
  header: string;
  key: string;
  inputValues: Array<{
    type?: 'string' | 'number';
    min?: number; // Define min and max for number input, if desired
    max?: number; // These values are ignored for string inputs
    cleanFunc?: (e: string) => string; // cleans a user input if needed
    displayStr: string;
    __value?: string; // For internal use only, please don't set this
  }>;
};

export type FilterPopoverOption =
  | FilterPopoverCheckboxOption
  | FilterPopoverInputOption;

export type FilterPopoverProps = {
  options: Array<FilterPopoverOption>;
  setFilter:
    | React.Dispatch<React.SetStateAction<string>>
    | ((value: string) => void);
};

function isCheckboxOption(x: FilterPopoverOption): boolean {
  return (x as FilterPopoverCheckboxOption).variant != null;
}
function isInputOption(x: FilterPopoverOption): boolean {
  return (x as FilterPopoverInputOption).inputValues != null;
}

function getCheckboxElement(
  option: FilterPopoverCheckboxOption,
): JSX.Element {
  const inputs = option.boxValues.map((val, i) => (
    <div key={i} className={styles.CheckDiv}>
      <input
        className={styles.Check}
        type={option.variant ?? 'checkbox'}
        defaultChecked={val.__checked}
        value={val.value}
        id={option.key + '-' + val.value}
        name={option.header}
        onClick={() => {
          if (option.variant === 'radio') {
            option.boxValues.forEach(
              (button) => (button.__checked = false),
            );
          }
          val.__checked = !val.__checked;
        }}
      />
      <label htmlFor={val.value}>{val.displayStr}</label>
    </div>
  ));
  const mid = Math.ceil(inputs.length / 2);

  return (
    <div className={styles.CheckboxRow}>
      <div className={styles.CheckboxColumn}>
        {inputs.slice(0, mid)}
      </div>
      <div className={styles.CheckboxColumn}>{inputs.slice(mid)}</div>
    </div>
  );
}

function getInputElement(
  option: FilterPopoverInputOption,
): JSX.Element {
  return (
    <Form>
      <Form.Group
        style={{
          display: option.inputValues.length === 1 ? 'block' : 'flex',
          marginBottom: '8px',
        }}
      >
        {option.inputValues.map((val, i) => (
          <div key={i} style={{ display: 'flex' }}>
            <Form.Control
              id={option.key}
              defaultValue={val.__value}
              onChange={(e) => (val.__value = e.target.value)}
              placeholder={val.displayStr}
              type={val.type}
              min={val.type === 'number' ? val.min : undefined}
              max={val.type === 'number' ? val.max : undefined}
              style={{ margin: '0px 2px' }}
            />
            {i < option.inputValues.length - 1 && (
              <div className={styles.DashContainer}>
                <p style={{ margin: 0 }}>&#8212;</p>
              </div>
            )}
          </div>
        ))}
      </Form.Group>
    </Form>
  );
}

const FilterPopover: React.FunctionComponent<FilterPopoverProps> = ({
  options,
  setFilter,
}: FilterPopoverProps) => {
  const applyFilter = () => {
    let output = '';
    options.forEach((option) => {
      let substr = `${option.key}=`;

      if (isCheckboxOption(option)) {
        (option as FilterPopoverCheckboxOption).boxValues.forEach(
          (val) => {
            if (val.__checked && val.value) {
              // Note: you can specify value = '' to make this
              // not run if you want no substr to be generated
              // This is useful, for example, if you have a boolean
              // decision like "has takeout", "does not have", or "any"
              if (substr.slice(-1) != '=') substr += ',';
              substr += val.value;
            }
          },
        );
      } else if (isInputOption(option)) {
        const cast = option as FilterPopoverInputOption;
        if (
          cast.inputValues.length == 2 &&
          cast.inputValues[0].type === 'number' &&
          cast.inputValues[1].type === 'number'
        ) {
          // NOTE: Assumed here that a 2 number input is a min/max pair
          const k = cast.key;
          const v1 = cast.inputValues[0].__value;
          const v2 = cast.inputValues[1].__value;
          substr = '';
          if (v1) substr += `min_${k}=${v1}`;
          if (v1 && v2) substr += '&';
          if (v2) substr += `max_${k}=${v2}`;
        } else {
          cast.inputValues.forEach((val) => {
            if (val.__value) {
              if (val.cleanFunc) substr += val.cleanFunc(val.__value);
              else substr += val.__value;
            }
          });
        }
      }

      if (substr.length > 0 && substr.slice(-1) != '=')
        output += '&' + substr;
    });
    if (output) setFilter(output);
  };

  const pop = (
    <Popover id="popover-basic" style={{ minWidth: '360px' }}>
      <Popover.Title>
        <div className={styles.TitleSplitter}>
          <h3>Filter Options</h3>

          {/*
           * This is a very unobvious way to close the popover
           * We make use of the rootClose parameter of the
           * OverlayTrigger and simulate a click onto the document
           * This click hits nothing but still closes the popover
           */}
          <Button
            aria-label="Close"
            onClick={() => {
              applyFilter();
              document.body.click();
            }}
          >
            Apply
          </Button>
        </div>
      </Popover.Title>
      <Popover.Content>
        {options.map((option, index) => {
          return (
            <div key={index}>
              <h4 className={styles.OptionHeader}>{option.header}</h4>

              {isCheckboxOption(option) &&
                getCheckboxElement(
                  option as FilterPopoverCheckboxOption,
                )}

              {isInputOption(option) &&
                getInputElement(option as FilterPopoverInputOption)}
            </div>
          );
        })}
      </Popover.Content>
    </Popover>
  );

  return (
    <div>
      <OverlayTrigger
        placement="bottom"
        overlay={pop}
        trigger="click"
        rootClose // Closes popover when clicking outside
      >
        <Button className={styles.PopoverButton}>
          <BsFilterCircle className={styles.IconAdjust} />
        </Button>
      </OverlayTrigger>
    </div>
  );
};

export default FilterPopover;
