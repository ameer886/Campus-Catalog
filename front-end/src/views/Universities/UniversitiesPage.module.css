.Universities {
  text-align: center;
}

.UniversitySplitter {
  display: flex;
}

.SplitterGrid {
  width: 75%;
  margin: 0 24px;
}

.SplitterInfo {
  width: 25%;
  margin-right: 12px;
}

.RadioButtonDiv {
  display: flex;
  align-items: center;
  margin-left: 8px;
}

.RadioButton {
  margin: 0 4px;
}
