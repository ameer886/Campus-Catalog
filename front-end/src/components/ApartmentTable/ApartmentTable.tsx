import React from 'react';

import { ColumnDefinitionType } from '../../components/GenericTable/GenericTable';
import PaginatedTable from '../../components/Pagination/PaginatedTable';
import { ApartmentRowType } from '../../views/Apartments/ApartmentsPage';

import './ApartmentTable.css';

type ApartmentTableProps = {
  rows: Array<ApartmentRowType>;
};

const apartmentTableHeaders: ColumnDefinitionType<
  ApartmentRowType,
  keyof ApartmentRowType
>[] = [
  {
    key: 'property_name',
    header: 'Property Name',
    sortFunc: (a, b) =>
      a.property_name.localeCompare(b.property_name),
  },
  {
    key: 'transit_score',
    header: 'Transit Score',
    sortFunc: (a, b) => {
      if (!a.transit_score) return -1;
      if (!b.transit_score) return 1;
      return a.transit_score - b.transit_score;
    },
  },
  {
    key: 'rating',
    header: 'Rating',
    sortFunc: (a, b) => {
      if (!a.rating) return -1;
      if (!b.rating) return 1;
      return a.rating - b.rating;
    },
  },
  {
    key: 'max_rent',
    header: 'Bedrooms',
    sortFunc: (a, b) => a.max_rent - b.max_rent,
  },
  {
    key: 'walk_score',
    header: 'Walk Score',
    sortFunc: (a, b) => {
      if (!a.walk_score) return -1;
      if (!b.walk_score) return 1;
      return a.walk_score - b.walk_score;
    },
  },
];

const ApartmentTable: React.FunctionComponent<ApartmentTableProps> =
  ({ rows }: ApartmentTableProps) => {
    return (
      <div className="ApartmentTable">
        <PaginatedTable
          columnDefinitions={apartmentTableHeaders}
          data={rows}
        />
      </div>
    );
  };

export default ApartmentTable;
