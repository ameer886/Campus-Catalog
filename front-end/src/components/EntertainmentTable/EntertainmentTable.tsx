import React from 'react';
import { useState } from 'react';

import type { ColumnDefinitionType } from '../../components/GenericTable/GenericTable';
import type { EntertainmentRowType } from '../../views/Entertainments/EntertainmentsPage';
import type { IntentionallyAny } from '../../utilities';
import type { FilterPopoverOption } from '../../components/FilterPopover/FilterPopover';

import GenericTable from '../GenericTable/GenericTable';
import PaginationRelay from '../Pagination/PaginationRelay';
import PaginatedTable from '../../components/Pagination/PaginatedTable';
import FilterPopover from '../../components/FilterPopover/FilterPopover';

import styles from './EntertainmentTable.module.css';

const entertainmentTableHeaders: ColumnDefinitionType<
  EntertainmentRowType,
  keyof EntertainmentRowType
>[] = [
  {
    key: 'amen_name',
    header: 'Amenity Name',
  },
  {
    key: 'city',
    header: 'City',
  },
  {
    key: 'state',
    header: 'State',
  },
  {
    key: 'num_review',
    header: 'Reviews',
  },
  {
    key: 'pricing',
    header: 'Price',
  },
  {
    key: 'rating',
    header: 'Rating',
  },
];

// This is exported for unit tests. You should not need to import it.
export const popoverOptions: FilterPopoverOption[] = [
  {
    header: 'City',
    key: 'city',
    inputValues: [
      {
        displayStr: 'Enter City here',
        // Capitalize only the first letter
        cleanFunc: (e) =>
          e
            .trim()
            .toLowerCase()
            .replace(/^\w/, (c) => c.toUpperCase()),
      },
    ],
  },
  {
    header: 'State',
    key: 'state',
    inputValues: [
      {
        displayStr: 'Enter state here (ex: TX)',
        cleanFunc: (e) => e.trim().toUpperCase(),
      },
    ],
  },
  {
    header: 'Rating',
    key: 'rate',
    variant: 'radio',
    boxValues: [
      { value: '0', displayStr: 'Any', __checked: true },
      { value: '1.0', displayStr: '>= 1.0' },
      { value: '2.0', displayStr: '>= 2.0' },
      { value: '3.0', displayStr: '>= 3.0' },
      { value: '4.0', displayStr: '>= 4.0' },
    ],
  },
  {
    header: 'Reviews',
    key: 'reviews',
    inputValues: [
      {
        displayStr: 'Minimum # of reviews',
        min: 0,
        type: 'number',
      },
    ],
  },
  {
    header: 'Price',
    key: 'price',
    variant: 'checkbox',
    boxValues: [
      { value: '$', displayStr: '$', __checked: true },
      { value: '$$', displayStr: '$$', __checked: true },
      { value: '$$$', displayStr: '$$$', __checked: true },
      { value: '$$$$', displayStr: '$$$$', __checked: true },
      { value: '$$$$$', displayStr: '$$$$$', __checked: true },
    ],
  },
  {
    header: 'Delivery',
    key: 'delivery',
    variant: 'radio',
    boxValues: [
      { value: 'True', displayStr: 'Has Delivery' },
      { value: 'False', displayStr: 'No Delivery' },
      { value: '', displayStr: 'Any', __checked: true },
    ],
  },
  {
    header: 'Takeout',
    key: 'takeout',
    variant: 'radio',
    boxValues: [
      { value: 'True', displayStr: 'Has Takeout' },
      { value: 'False', displayStr: 'No Takeout' },
      { value: '', displayStr: 'Any', __checked: true },
    ],
  },
];

// This is an optional property to ONLY BE USED IN JEST TESTS
// PLEASE do not use testRows in production
type EntertainmentTableTestProps = {
  testRows?: Array<EntertainmentRowType>;
};

const EntertainmentTable: React.FunctionComponent<EntertainmentTableTestProps> =
  ({ testRows }: EntertainmentTableTestProps) => {
    if (testRows) {
      // Hack around queries to get a "table" that tests the basic fns
      // We need to avoid any kind of query while keeping tests functional
      // If testRows is not undefined, then this will most certainly
      // crash the front-end
      const ps = 10;
      const [filter, setFilter] = useState('');
      const [sortStr, setSortStr] = useState('NONE');
      const [page, setPage] = useState(1);

      return (
        <div className={styles.EntertainmentTable}>
          <div>
            <div className={styles.FilterButton}>
              <FilterPopover
                options={popoverOptions}
                setFilter={(e: string) => {
                  if (filter === e) return;
                  setFilter(e);
                }}
              />
            </div>
          </div>

          <GenericTable
            columnDefinitions={entertainmentTableHeaders}
            data={testRows.slice((page - 1) * ps, page * ps)}
            parentSort={setSortStr}
            parentStr={sortStr}
          />

          <PaginationRelay
            curPage={page}
            setPage={setPage}
            pageSize={ps}
            totalElements={testRows.length}
          />
        </div>
      );
    }

    const processResponse = (data: IntentionallyAny) => {
      const responseRows = data[1].amenities.map(
        (amen: IntentionallyAny) => {
          return {
            id: amen.amen_id,
            ...amen,
          };
        },
      );
      return responseRows;
    };

    return (
      <div className={styles.EntertainmentTable}>
        <PaginatedTable
          options={popoverOptions}
          processResponse={processResponse}
          model="amenities"
          columnDefinitions={entertainmentTableHeaders}
        />
      </div>
    );
  };

export default EntertainmentTable;
