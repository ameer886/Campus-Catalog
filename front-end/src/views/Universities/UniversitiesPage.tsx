import React, { useEffect } from 'react';
import { useState } from 'react';
import styles from './UniversitiesPage.module.css';

import type {
  Address,
  AmenityKey,
  PropertyKey,
} from '../../universalTypes';
import type { PaginationMeta } from '../../components/Pagination/PaginatedTable';
import type { IntentionallyAny } from '../../utilities';
import type { FilterPopoverOption } from '../../components/FilterPopover/FilterPopover';

import FilterPopover from '../../components/FilterPopover/FilterPopover';
import UniversityGrid from '../../components/UniversityGrid/UniversityGrid';
import PaginationRelay from '../../components/Pagination/PaginationRelay';

import { getAPI } from '../../APIClient';

const PAGE_SIZE = 9;

// Type of a single university
export type UniversityType = {
  acceptance_rate?: number;
  alias: string | null;
  avg_cost_attendance?: number;
  avg_sat?: number;
  carnegie_undergrad: string;
  graduation_rate?: number;
  image: string;
  latitude: number;
  locale: string;
  location: Address;
  longitude: number;
  num_graduate?: number;
  num_undergrad?: number;
  ownership_id: string;
  rank: number;
  school_url: string;
  tuition_in_st?: number;
  tuition_out_st?: number;
  univ_id: string;
  univ_name: string;
  zip_code: string;

  amenities_nearby: Array<AmenityKey>;
  housing_nearby: Array<PropertyKey>;
};

export type UniversityRowType = {
  acceptance_rate: number;
  avg_cost_attendance: number;
  city: string;
  ownership_id: string;
  rank: number;
  state: string;
  tuition_in_st: number;
  tuition_out_st: number;
  univ_id: string;
  univ_name: string;

  id: string;
};

const popoverOptions: FilterPopoverOption[] = [
  {
    header: 'City',
    key: 'city',
    inputValues: [
      {
        displayStr: 'Enter City here',
        // Capitalize only the first letter
        cleanFunc: (e) =>
          e
            .trim()
            .toLowerCase()
            .replace(/^\w/, (c) => c.toUpperCase()),
      },
    ],
  },
  {
    header: 'State',
    key: 'state',
    inputValues: [
      {
        displayStr: 'Enter state here (ex: TX)',
        cleanFunc: (e) => e.trim().toUpperCase(),
      },
    ],
  },
  {
    header: 'Acceptance Rate',
    key: 'accept',
    inputValues: [
      {
        displayStr: 'Minimum Acceptance %',
        type: 'number',
        min: 0,
        max: 100,
        cleanFunc: (e) => (parseInt(e) / 100.0).toFixed(2),
      },
    ],
  },
  {
    header: 'Graduation Rate',
    key: 'grad',
    inputValues: [
      {
        displayStr: 'Minimum Graduation %',
        type: 'number',
        min: 0,
        max: 100,
        cleanFunc: (e) => (parseInt(e) / 100.0).toFixed(2),
      },
    ],
  },
  {
    header: 'Ownership',
    key: 'ownership_id',
    variant: 'radio',
    boxValues: [
      { displayStr: 'Any', value: '', __checked: true },
      { displayStr: 'Public', value: '1' },
      { displayStr: 'Private Non-Profit', value: '2' },
      { displayStr: 'Private For-Profit', value: '3' },
    ],
  },
];

const inputPairs = [
  { displayStr: 'School Name', sortOrder: 'univ_name' },
  { displayStr: 'Rank', sortOrder: 'rank' },
  { displayStr: 'City', sortOrder: 'city' },
  { displayStr: 'State', sortOrder: 'state' },
  { displayStr: 'In-State Tuition', sortOrder: 'tuition_in_st' },
  { displayStr: 'Out-of-State Tuition', sortOrder: 'tuition_out_st' },
  { displayStr: 'Acceptance Rate', sortOrder: 'acceptance_rate' },
];

/*
 * Type to create a single university input button (radio button thing)
 * sortOrder is the sortOrder string that button selects
 * displayStr is the label of the button
 * defaultChecked should be set to true to make a button automatically be selected
 * onChange should just really be the setSortOrder function
 */
type UniversityInputProps = {
  sortOrder: string;
  displayStr: string;
  defaultCheckedFunc?: (mySort: string) => boolean;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
};

/*
 * Build a single radio button for one of the sort orders
 */
const UniversityInput: React.FunctionComponent<UniversityInputProps> =
  ({
    sortOrder,
    displayStr,
    defaultCheckedFunc,
    onChange,
  }: UniversityInputProps) => {
    return (
      <div>
        <div className={styles.RadioButtonDiv}>
          <input
            className={styles.RadioButton}
            type="radio"
            value={sortOrder}
            id={sortOrder}
            name="sortOrder"
            defaultChecked={
              defaultCheckedFunc
                ? defaultCheckedFunc(sortOrder)
                : false
            }
            onChange={(e) => onChange(e)}
          />
          <label htmlFor={sortOrder}>{displayStr}</label>
        </div>
      </div>
    );
  };

const UniversityInputPair: React.FunctionComponent<UniversityInputProps> =
  ({ sortOrder, displayStr, ...props }: UniversityInputProps) => {
    return (
      <>
        <UniversityInput
          displayStr={displayStr + ' (Ascending)'}
          sortOrder={sortOrder + '_asc'}
          {...props}
        />
        <UniversityInput
          displayStr={displayStr + ' (Descending)'}
          sortOrder={sortOrder + '_dsc'}
          {...props}
        />
      </>
    );
  };

/*
 * The Universities page
 * One of the three main model collection pages
 * Should contain a list of universities in a sortable table/grid
 */
const UniversitiesPage: React.FunctionComponent = () => {
  const [sortOrder, setSortOrder] = useState('NONE');
  const [page, setPage] = useState(1);
  const [loading, setLoading] = useState(true);
  const [meta, setMeta] = useState<PaginationMeta | null>(null);
  const [cards, setCards] = useState<Array<UniversityRowType>>([]);
  const [filter, setFilter] = useState('');

  useEffect(() => {
    const fetchDataAsync = async () => {
      try {
        let params = `page=${page}&per_page=${PAGE_SIZE}`;
        if (sortOrder !== 'NONE') {
          params += `&sort=${sortOrder.slice(0, -4)}`;
          if (sortOrder.includes('dsc')) params += '&desc=True';
        }
        if (filter) params += filter;
        console.log(params);

        const data = await getAPI({
          model: 'universities',
          params: params,
        });
        const responseMeta: PaginationMeta = {
          ...data[0],
        };
        const responseCards = data[1].universities
          .map((university: IntentionallyAny) => {
            return {
              id: university.univ_id,
              ...university,
            };
          })
          .filter(
            (university: IntentionallyAny) => university.rank != null,
          );
        setCards(responseCards);
        setMeta(responseMeta);
        setLoading(false);
      } catch (err) {
        console.error(err);
      }
    };
    fetchDataAsync();
  }, [page, filter, sortOrder]);

  // Note: ALL HOOKS must come before a return
  if (loading || meta == null)
    return (
      <div style={{ textAlign: 'center' }}>
        <h1>Universities</h1>
        <p>Loading responses, please be patient.</p>
      </div>
    );

  // wrapper to change the sort order of the cards
  const updateSort = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSortOrder(e.target.value);
    setPage(1);
    setLoading(true);
  };

  return (
    <div className={styles.Universities}>
      <h1>Universities</h1>
      <div className={styles.UniversitySplitter}>
        {/* Build the grid on the left */}
        <div className={styles.SplitterGrid}>
          <UniversityGrid cards={cards} />
          <PaginationRelay
            curPage={page}
            setPage={(e) => {
              setLoading(true);
              setPage(e);
            }}
            pageSize={PAGE_SIZE}
            totalElements={meta.total_items}
          />
        </div>

        {/* Build the inputs on the right */}
        <div className={styles.SplitterInfo}>
          <FilterPopover
            options={popoverOptions}
            setFilter={(e) => {
              setPage(1);
              setLoading(true);
              setFilter(e);
            }}
          />
          <p style={{ marginBottom: '0' }}>
            Select your desired sort order:
          </p>
          <form>
            {/* Default sort: do nothing */}
            <UniversityInput
              displayStr="None"
              sortOrder="NONE"
              defaultCheckedFunc={(s) => s === sortOrder}
              onChange={updateSort}
            />

            {/* All pairs */}
            {inputPairs.map((input, index) => (
              <UniversityInputPair
                key={index}
                displayStr={input.displayStr}
                sortOrder={input.sortOrder}
                onChange={updateSort}
                defaultCheckedFunc={(s) => s === sortOrder}
              />
            ))}
          </form>
        </div>
      </div>
    </div>
  );
};

export default UniversitiesPage;
