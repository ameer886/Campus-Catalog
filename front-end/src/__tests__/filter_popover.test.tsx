import React from 'react';

import { render, screen, fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FilterPopover from '../components/FilterPopover/FilterPopover';

import { popoverOptions as housingOptions } from '../components/ApartmentTable/ApartmentTable';
import { popoverOptions as amenityOptions } from '../components/EntertainmentTable/EntertainmentTable';
import { popoverOptions as univOptions } from '../views/Universities/UniversitiesPage';

configure({ adapter: new Adapter() });

describe('Filter Popover Test Suite', () => {
  it('snapshot test: housing options', () => {
    const snap = shallow(
      <FilterPopover
        options={housingOptions}
        setFilter={() => {
          return;
        }}
      />,
    );
    expect(snap).toMatchSnapshot();
  });

  it('snapshot test: amenity options', () => {
    const snap = shallow(
      <FilterPopover
        options={amenityOptions}
        setFilter={() => {
          return;
        }}
      />,
    );
    expect(snap).toMatchSnapshot();
  });

  it('snapshot test: university options', () => {
    const snap = shallow(
      <FilterPopover
        options={univOptions}
        setFilter={() => {
          return;
        }}
      />,
    );
    expect(snap).toMatchSnapshot();
  });

  /*
  it('input test: basic string', () => {
    let s = '';
    const setS = (value: React.SetStateAction<string>) => {
      s = value.toString();
    };
    render(
      <FilterPopover
        setFilter={setS}
        options={[
          {
            header: 'Basic String',
            key: 'key',
            inputValues: [
              {
                displayStr: 'Placeholder Text',
              },
            ],
          },
        ]}
      />,
    );

    openPopover();
    const textInput = screen.getByRole('textbox') as HTMLInputElement;
    expect(textInput).not.toBeNull();

    const applyButton = screen.getByRole('button', { name: 'Close' });
    expect(applyButton).not.toBeNull();

    textInput.value = 'value';
    fireEvent.click(applyButton);

    expect(s).toEqual('test input');
  });
  */
});
