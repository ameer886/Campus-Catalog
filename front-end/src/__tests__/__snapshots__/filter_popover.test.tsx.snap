// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Popover Test Suite snapshot test: amenity options 1`] = `ShallowWrapper {}`;

exports[`Filter Popover Test Suite snapshot test: housing options 1`] = `ShallowWrapper {}`;

exports[`Filter Popover Test Suite snapshot test: university options 1`] = `ShallowWrapper {}`;
